### DISCUSSION TITLE
Is it possible for a formula to create directories in the user's home directory?

### DISCUSSION THREAD

**Discussion by davidji99** [Homebrew/discussions#778] [Tap maintenance and Homebrew development] [answered]
I'm exploring a way of using `brew` to install private terraform providers binaries. These binaries must be installed in a specific directory in `$HOME`. So far my attempts to create a directory in the formula using `mkdir` or `system` have been met with `Operation Not Permitted`. I'd like to know if this is all possible to do with brew?

I'm using macOS 10.15.7.

**Comment by carlocab** [comment, marked as answer]
If these are pre-built binaries, it might be better to write a cask rather than a formula. If you must use a formula, you might be able to write to `$HOME` in a `post_install` block. `install` blocks are run in a sandbox, which prevents installers from writing wherever they want to in your system.

**Reply by Bo98** [reply]
I also suggest investigating whether symlinks will work rather than actually installing things into `$HOME`.

**Reply by davidji99** [reply]
Using a Cask works in the sense that I can use `mv` to move a file from the app directory to somewhere in `$HOME`. It's a bit of a hack getting this to work.

`post_install` still doesn't have the ability to effect change in `$HOME`.
